api/src/main/java/com/viaversion/viaversion/api/minecraft/entities/EntityTypes1_13.java

```diff
@@ -29,201 +29,194 @@
-
-        // Zombies - META CHECKED
-        ZOMBIE(87, ABSTRACT_MONSTER), // aki
-        DROWNED(14, ZOMBIE), // ajg
-        HUSK(30, ZOMBIE), // ajp
-        ZOMBIE_PIGMAN(53, ZOMBIE), // aju
-        ZOMBIE_VILLAGER(89, ZOMBIE), // akj
+        AREA_EFFECT_CLOUD(0, ENTITY),
+        END_CRYSTAL(16, ENTITY),
+        EVOKER_FANGS(20, ENTITY),
+        EXPERIENCE_ORB(22, ENTITY),
+        EYE_OF_ENDER(23, ENTITY),
+        FALLING_BLOCK(24, ENTITY),
+        ITEM(32, ENTITY),
+        TNT(55, ENTITY),
+        LIGHTNING_BOLT(91, ENTITY), // Weather effect entity
 
-        // Flying entities
-        ABSTRACT_FLYING(-1, ABSTRACT_INSENTIENT), // abt
-        GHAST(26, ABSTRACT_FLYING), // ajm
-        PHANTOM(90, ABSTRACT_FLYING), // ajt
-
-        ABSTRACT_AMBIENT(-1, ABSTRACT_INSENTIENT), // afy
-        BAT(3, ABSTRACT_AMBIENT), // afz
-
-        ABSTRACT_WATERMOB(-1, ABSTRACT_INSENTIENT), // agx
-        SQUID(70, ABSTRACT_WATERMOB), // agt
-        DOLPHIN(12, ABSTRACT_WATERMOB),
-
-        // Slimes
-        SLIME(64, ABSTRACT_INSENTIENT), // aka
-        MAGMA_CUBE(38, SLIME), // ajr
-
-        // Hangable objects
-        ABSTRACT_HANGING(-1, ENTITY), // aim
-        LEASH_KNOT(35, ABSTRACT_HANGING), // aio
-        ITEM_FRAME(33, ABSTRACT_HANGING), // ain
-        PAINTING(49, ABSTRACT_HANGING), // aiq
-
-        ABSTRACT_LIGHTNING(-1, ENTITY), // aiu
-        LIGHTNING_BOLT(91, ABSTRACT_LIGHTNING), // aiv
-
-        // Arrows
-        ABSTRACT_ARROW(-1, ENTITY), // akw
-        ARROW(2, ABSTRACT_ARROW), // aky
-        SPECTRAL_ARROW(68, ABSTRACT_ARROW), // alk
-        TRIDENT(94, ABSTRACT_ARROW), // alq
-
-        // Fireballs
-        ABSTRACT_FIREBALL(-1, ENTITY), // akx
-        DRAGON_FIREBALL(13, ABSTRACT_FIREBALL), // akz
-        FIREBALL(34, ABSTRACT_FIREBALL), // ald
-        SMALL_FIREBALL(65, ABSTRACT_FIREBALL), // ali
-        WITHER_SKULL(85, ABSTRACT_FIREBALL), // alr
+        // Hanging entities
+        HANGING_ENTITY(ENTITY),
+        LEASH_KNOT(35, HANGING_ENTITY),
+        ITEM_FRAME(33, HANGING_ENTITY),
+        PAINTING(49, HANGING_ENTITY),
 
         // Projectiles
-        PROJECTILE_ABSTRACT(-1, ENTITY), // all
-        SNOWBALL(67, PROJECTILE_ABSTRACT), // alj
-        ENDER_PEARL(75, PROJECTILE_ABSTRACT), // aln
-        EGG(74, PROJECTILE_ABSTRACT), // alm
-        POTION(77, PROJECTILE_ABSTRACT), // alp
-        EXPERIENCE_BOTTLE(76, PROJECTILE_ABSTRACT), // alo
+        PROJECTILE(ENTITY),
+        FIREWORK_ROCKET(25, ENTITY),
+        LLAMA_SPIT(37, ENTITY),
+        SHULKER_BULLET(60, ENTITY),
+        SNOWBALL(67, PROJECTILE),
+        ENDER_PEARL(75, PROJECTILE),
+        EGG(74, PROJECTILE),
+        EXPERIENCE_BOTTLE(76, PROJECTILE),
+        POTION(77, PROJECTILE),
+        FISHING_BOBBER(93, ENTITY),
+
+        ABSTRACT_ARROW(ENTITY),
+        ARROW(2, ABSTRACT_ARROW),
+        SPECTRAL_ARROW(68, ABSTRACT_ARROW),
+        TRIDENT(94, ABSTRACT_ARROW),
+
+        ABSTRACT_FIREBALL(ENTITY),
```

Comment:
Note that this is a bad mcp name, these aren't fireballs but Mojang `AbstractHurtingProjectile`, or later fixed in mcp as `DamagingProjectileEntity`

Reply:
They have the same name in modern enums, should I change all namings then?

Reply:
There's no hurting projectile type in those, that's the *actual* abstract fireball, which was added in 1.14

Reply:
Ah I see, changed the name to HURTING_PROJECTILE